gossip/src/duplicate_shred_handler.rs

```diff
@@ -85,7 +97,12 @@ impl DuplicateShredHandler {
         self.last_root = last_root;
         if let Ok(bank_fork) = self.bank_forks.try_read() {
             let root_bank = bank_fork.root_bank();
+            self.cached_epoch_schedule = Some(root_bank.epoch_schedule().clone());
```

Comment:
Does this need to be cloned everytime? I don't think `epoch_schedule` varies epoch to epoch, despite the name. Might be better initizlied in the constructor from the root bank in `bank_forks`: https://github.com/solana-labs/solana/blob/61ce88a2ab6f8201baaf205e695a3045d234b1dd/gossip/src/duplicate_shred_handler.rs#L73

Reply:
good catch